core/server/master/src/main/java/alluxio/master/file/meta/InodeLockManager.java

```diff
@@ -0,0 +1,168 @@
+   * cache to get the appropriate read lock.
+   *
+   * We use weak values so that when nothing holds a reference to
+   * a lock, the garbage collector can remove the lock's entry from the cache.
+   */
+  public final LoadingCache<Long, WeakSafeReentrantReadWriteLock> mInodeLocks =
+      CacheBuilder.<Long, WeakSafeReentrantReadWriteLock>newBuilder()
+          .weakValues()
+          .initialCapacity(1_000)
+          .concurrencyLevel(100)
+          .build(new CacheLoader<Long, WeakSafeReentrantReadWriteLock>() {
+            @Override
+            public WeakSafeReentrantReadWriteLock load(Long key) {
+              return new WeakSafeReentrantReadWriteLock();
+            }
+          });
+
+  /**
+   * Cache for supplying edge locks, similar to mInodeLocks.
+   */
+  public final LoadingCache<Edge, WeakSafeReentrantReadWriteLock> mEdgeLocks =
+      CacheBuilder.<Long, WeakSafeReentrantReadWriteLock>newBuilder()
+          .weakValues()
+          .initialCapacity(1_000)
+          .concurrencyLevel(100)
+          .build(new CacheLoader<Edge, WeakSafeReentrantReadWriteLock>() {
+            @Override
+            public WeakSafeReentrantReadWriteLock load(Edge key) {
+              return new WeakSafeReentrantReadWriteLock();
+            }
+          });
+
+  /**
+   * Cache for supplying inode persistence locks. Before a thread can persist an inode, it must
+   * acquire the persisting lock for the inode. The cache maps inode ids to AtomicBooleans used to
+   * provide mutual exclusion for inode persisting threads.
+   */
+  private final LoadingCache<Long, AtomicBoolean> mPersistingLocks =
+      CacheBuilder.newBuilder()
+          .weakValues()
+          .initialCapacity(1_000)
+          .concurrencyLevel(100)
+          .build(new CacheLoader<Long, AtomicBoolean>() {
+            @Override
+            public AtomicBoolean load(Long key) {
+              return new AtomicBoolean();
+            }
+          });
+
+  @VisibleForTesting
+  boolean inodeReadLockedByCurrentThread(long inodeId) {
+    return mInodeLocks.getUnchecked(inodeId).getReadHoldCount() > 0;
+  }
+
+  @VisibleForTesting
+  boolean inodeWriteLockedByCurrentThread(long inodeId) {
+    return mInodeLocks.getUnchecked(inodeId).getWriteHoldCount() > 0;
+  }
+
+  @VisibleForTesting
+  boolean edgeReadLockedByCurrentThread(Edge edge) {
+    return mEdgeLocks.getUnchecked(edge).getReadHoldCount() > 0;
+  }
+
+  @VisibleForTesting
+  boolean edgeWriteLockedByCurrentThread(Edge edge) {
+    return mEdgeLocks.getUnchecked(edge).getWriteHoldCount() > 0;
+  }
+
+  /**
+   * Acquires an inode lock.
+   *
+   * @param inode the inode to lock
+   * @param mode the mode to lock in
+   * @return a lock resource which must be closed to release the lock
+   */
+  public LockResource lockInode(InodeView inode, LockMode mode) {
+    switch (mode) {
+      case READ:
+        return new LockResource(mInodeLocks.getUnchecked(inode.getId()).readLock());
```

Comment:
Use a similar code pattern as `lockEdge`?

Reply:
thanks, refactored the common code